crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -969,13 +1001,18 @@ struct EntityMeta {
     pub generation: NonZero<u32>,
     /// The current location of the [`Entity`]
     pub location: EntityLocation,
+    /// Location of the last spawn or despawn of this entity
+    #[cfg(feature = "track_change_detection")]
+    spawned_despawned_by: &'static Location<'static>,
```

Comment:
I experimented with this change as well, but noticed this safety comment on `EntityMeta`:

> Safety:
> This type must not contain any pointers at any level, and be safe to fully fill with `u8::MAX`.

I believe this new field breaks _both_ of these safety invariants, since it's a pointer type (static reference), _and_ it's not safe to fill with all ones.

I _do_ think it _should_ be possible to relax the pointer requirement, since this is a read-only static reference (I can't imagine a scenario where some combination of transmuting, dropping, etc. could be unsound). But the all-ones-fill could create an invalid reference.

I think the most transparent workaround here would be to use a `union` type:

```rust
#[repr(C)]
union LocationOrByteFill {
    fill: NonZero<usize>,
    location: &'static Location<'static>,
}
```

Then when reading the value, you'd have to check if the `fill` value is `usize::MAX`, and if so, panic.

Alternatively, maybe you could update `flush_and_reserve_invalid_assuming_no_entities` (the function that performs the fill) to also track-caller and use the real `Location` value. I'm not sure which is easier/more performant.

Reply:
Could potentially remove the method. I don't think it's being used anymore after retained render world

Reply:
With the safety comment updated that would also be an acceptable resolution to the issue.